imagepipeline-base/src/main/java/com/facebook/imageformat/ImageFormat.java

```diff
@@ -32,7 +35,8 @@
      * @return the image format or null if unknown
      */
     @Nullable
-    ImageFormat determineFormat(byte[] headerBytes, int headerSize);
+    ImageFormat determineFormat(byte[] headerBytes, int headerSize, InputStream is)
```

Comment:
Got it. Some of the first part of your message got clipped somehow, but I think I see where you were going with it.

Knowing beforehand which images will be SVG, JPEG, etc. at this point in the code would be nice, but the first question that comes to mind is, "from where do we get that information?" At some point or another, the header of the file will have to be searched to determine the file type. If it's here, great. If it's "before" this point in the workflow, that's fine as well, but until we discover an additional advantage (preferably several others) to having the header-parsing code somewhere other than here, it's probably not worth the effort to move the code.

Maybe the API user could be given the option to provide image types for all of their images. I would be a significant change, but it's certainly interesting to think about.

For now, I'll set up some simple Exception messages like I mentioned in option 2 of my last comment and then I'll leave the merge decision up to you guys.